Game/Source/GDKTestGyms/BenchmarkGymGameModeBase.cpp

```diff
@@ -852,78 +880,91 @@ void ABenchmarkGymGameModeBase::InitMemReportTimer(const FString& MemReportInter
 }
 #endif
 
-int32 ABenchmarkGymGameModeBase::GetPlayerControllerCount() const
+void ABenchmarkGymGameModeBase::ReportAuthoritativeActorCount_Implementation(const int32 ActorCountId, const FString& WorkerID, const TArray<FActorCount>& ActorCounts)
 {
-	int32 Count = 0;
-	for (FConstPlayerControllerIterator PCIt = GetWorld()->GetPlayerControllerIterator(); PCIt; ++PCIt)
+	ActorCountMap& ActorCountMap = WorkerActorCounts.FindOrAdd(WorkerID);
+	for (const FActorCount& ActorCount : ActorCounts)
+	{
+		ActorCountMap.FindOrAdd(ActorCount.ActorClass) = ActorCount.Count;
+	}
+
+	ActorCountReportedIds.FindOrAdd(WorkerID) = ActorCountId;
+	if (ActorCountReportedIds.Num() == NumWorkers)
 	{
-		if (APlayerController* PC = PCIt->Get())
+		bool bAllWorkersInSync = true;
+		for (const auto& Pair : ActorCountReportedIds)
 		{
-			if (PC->HasAuthority())
-			{
-				++Count;
-			}
-			else if (USpatialNetDriver* SpatialDriver = Cast<USpatialNetDriver>(GetNetDriver()))
+			if (Pair.Value != ActorCountReportId)
 			{
-				// During actor authority handover, there's a period where no server will believe it has authority over
-				// the Unreal actor, but will still have authority over the entity. To better minimize this period, use
-				// the spatial authority as a fallback validation.
-				Worker_EntityId EntityId = SpatialDriver->PackageMap->GetEntityIdFromObject(PC);
-				const SpatialGDK::EntityViewElement* Element = SpatialDriver->Connection->GetView().Find(EntityId);
-				if (Element != nullptr && Element->Authority.Contains(SpatialConstants::SERVER_AUTH_COMPONENT_SET_ID))
-				{
-					++Count;
-				}
+				bAllWorkersInSync = false;
+				break;
 			}
 		}
+
+
+		ActorCountReportedIds.Empty();
+		if (bAllWorkersInSync)
```

Comment:
Little concerned that if workers get desynced in some way, this will never fire, and we'll get a potential false positive. Thoughts?

Reply:
As in, if we are always desynced, will never check the actors count is invalid so the actor count metric wont fail so the test would pass. 

I think we do need to ensure the workers are in sync, otherwise there is no point looking at the reported values.
It's worth pointing out if there is a desync, subsequent actor counts should not be affected unless there are continuous desyncs or very frequent desyncs.  

We could fail the test though if we see desyncs over a threshold. It would be nice to actually see how often this case occurs.

Reply:
Yeah I think we should just add a timer that's perhaps 2x the actor sync reporting time, and each time the auth server processes `UpdateAndCheckTotalActorCounts` we reset the timer. If it ever times out, it can fail the actor count metric but with a custom reason why.

Reply:
Custom reason being a custom log or a new metric/ expectation?